packages/sdk/src/utils/getChainId.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  JsonRpcProvider,
+  BrowserProvider,
+  AbstractProvider,
+  AbstractSigner,
+  Eip1193Provider
+} from 'ethers';
+import { DEFAULT_CHAIN_ID } from '../config/config.js';
+
+type EthersCompatibleProvider =
+  | string
+  | AbstractProvider
+  | AbstractSigner
+  | Eip1193Provider;
+
+export async function getChainIdFromProvider(ethProvider: EthersCompatibleProvider): Promise<number> {
+  try {
+    if (typeof ethProvider === 'string') {
+      const provider = new JsonRpcProvider(ethProvider);
+      const network = await provider.getNetwork();
+      return Number(network.chainId);
+    } else if ('request' in ethProvider) {
+      const provider = new BrowserProvider(ethProvider as Eip1193Provider);
+      const network = await provider.getNetwork();
+      return Number(network.chainId);
+    } else if ('getNetwork' in ethProvider) {
+      const network = await (ethProvider as AbstractProvider).getNetwork();
+      return Number(network.chainId);
+    }
```

Comment:
- We can avoid using `"getNetwork" in ethProvider` check by checking ethers class instance, which is more restrictive
- AbstractSigner does not implement `getNetwork`, we need to use the connected provider if it exists
- `'request' in ethProvider` to test the EIP1193 case is the less restrictive check, and could have false positive matches,  we should perform it last 

```suggestion
    } else if (ethProvider instanceof AbstractProvider) {
      const network = await ethProvider.getNetwork();
      return Number(network.chainId);
    } else if (ethProvider instanceof AbstractSigner) {
      const { provider } = ethProvider;
      if (!provider) {
        throw Error('Signer is not connected to a provider');
      }
      const network = await provider.getNetwork();
      return Number(network.chainId);
    } else if ('request' in ethProvider) {
      const provider = new BrowserProvider(ethProvider as Eip1193Provider);
      const network = await provider.getNetwork();
      return Number(network.chainId);
    }
```